core/src/banking_stage/consumer.rs

```diff
@@ -392,14 +392,54 @@ impl Consumer {
         bank: &Arc<Bank>,
         txs: &[SanitizedTransaction],
         chunk_offset: usize,
+    ) -> ProcessTransactionBatchOutput {
+        // No filtering before QoS - transactions should have been sanitized immediately prior to this call
+        let pre_results = std::iter::repeat(Ok(()));
+        self.process_and_record_transactions_with_pre_results(bank, txs, chunk_offset, pre_results)
+    }
+
+    pub fn process_and_record_aged_transactions(
+        &self,
+        bank: &Arc<Bank>,
+        txs: &[SanitizedTransaction],
+        max_slot_ages: &mut [Slot],
+    ) -> ProcessTransactionBatchOutput {
+        // Need to filter out transactions since they were sanitized earlier.
+        // This means that the transaction may cross and epoch boundary (not allowed),
+        //  or account lookup tables may have been closed.
+        let pre_results = txs
+            .iter()
+            .zip(max_slot_ages)
+            .map(|(tx, max_slot_age)| {
+                if *max_slot_age < bank.slot() {
+                    *max_slot_age = 0; // Sentinal value to indicate transaction was sanitized before epoch boundary
```

Comment:
nit: how about introducing some constant like `PREV_EPOCH_SENTINEL_SLOT`?

better yet, i like small custom type like this:

```rust
struct MaxAgeInSlot(Slot);

impl MaxAgeInSlot {
    fn mark_as_sanitized_at_prev_epoch(&mut self) {
    }
}
```

or ideally, `enum` (this incurs some increased memory footprint, though..)

Reply:
I think following the other [comment](https://github.com/solana-labs/solana/pull/30970/files/edbf4f67ca662f9e86f62b12ce668d351ffd7343#r1194648995), we can entirely remove the need of a sentinel value.

Reply:
I think by doing the re-sanitization on consume-side, we can avoid needing the wrapper type at all. wdyt?